src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/customcubic/CustomGeneratorSettingsFixer.java

```diff
@@ -0,0 +1,534 @@
+            newRoot.add("lowNoiseFrequencyZ", getLowNoiseFrequencyZ(oldRoot));
+        if (!calledForCubeArea || oldRoot.has("lowNoiseOctaves"))
+            newRoot.add("lowNoiseOctaves", getLowNoiseOctaves(oldRoot));
+        if (!calledForCubeArea || oldRoot.has("highNoiseFactor"))
+            newRoot.add("highNoiseFactor", getHighNoiseFactor(oldRoot));
+        if (!calledForCubeArea || oldRoot.has("highNoiseOffset"))
+            newRoot.add("highNoiseOffset", getHighNoiseOffset(oldRoot));
+        if (!calledForCubeArea || oldRoot.has("highNoiseFrequencyX"))
+            newRoot.add("highNoiseFrequencyX", getHighNoiseFrequencyX(oldRoot));
+        if (!calledForCubeArea || oldRoot.has("highNoiseFrequencyY"))
+            newRoot.add("highNoiseFrequencyY", getHighNoiseFrequencyY(oldRoot));
+        if (!calledForCubeArea || oldRoot.has("highNoiseFrequencyZ"))
+            newRoot.add("highNoiseFrequencyZ", getHighNoiseFrequencyZ(oldRoot));
+        if (!calledForCubeArea || oldRoot.has("highNoiseOctaves"))
+            newRoot.add("highNoiseOctaves", getHighNoiseOctaves(oldRoot));
+        if (!calledForCubeArea)
+            newRoot.add("cubeAreas", getCubeAreas(oldRoot));
+        if (!calledForCubeArea || oldRoot.has("replacerConfig"))
+            newRoot.add("replacerConfig", getReplacerConfig(oldRoot));
+
+        String newGeneratorOptions = gson.toJson(newRoot).replaceAll("cubicchunks:", MODID + ":");
+        return newGeneratorOptions;
+    }
+
+    private static JsonElement getReplacerConfig(JsonObject json) {
+        JsonReader reader = new JsonReader(new StringReader("{\"defaults\":{\"cubicgen:biome_fill_noise_octaves\":4.0,\"cubicgen:ocean_block\":{\"Properties\":{\"level\":\"0\"},\"Name\":\"minecraft:water\"},\"cubicgen:height_scale\":64.0,\"cubicgen:biome_fill_noise_freq\":0.0078125,\"cubicgen:water_level\":63.0,\"cubicgen:biome_fill_depth_factor\":2.3333333333333335,\"cubicgen:terrain_fill_block\":{\"Properties\":{\"variant\":\"stone\"},\"Name\":\"minecraft:stone\"},\"cubicgen:mesa_depth\":16.0,\"cubicgen:biome_fill_depth_offset\":3.0,\"cubicgen:horizontal_gradient_depth_decrease_weight\":1.0,\"cubicgen:height_offset\":64.0},\"overrides\":{}}"));
+        JsonObject biomeBlockReplacerConfigDefaultJson = new JsonParser().parse(reader).getAsJsonObject();
+        return getOrDefault(json, "replacerConfig", biomeBlockReplacerConfigDefaultJson);
+    }
+
+    private static JsonElement getCubeAreas(JsonObject json) {
+        JsonArray cubeAreas = new JsonArray();
+        if (json.has("cubeAreas") && json.get("cubeAreas").isJsonArray()) {
+            JsonArray array = json.get("cubeAreas").getAsJsonArray();
+            for (JsonElement entry : array) {
+                JsonArray mapEntry = entry.getAsJsonArray();
+                JsonElement key = mapEntry.get(0);
+                JsonObject value = stringToJson(fixGeneratorOptions(mapEntry.get(1).getAsJsonObject(), true));
```

Comment:
This needs the "parent" one as context, and use parent preset as defaults (this also means `getCubeAreas` needs to be called last, so that the parent preset is otherwise complete)

Reply:
This means that every cube area will be as huge as full list of settings. And manual editing of preset would be slightly harder. Are you sure it should be that way?

Reply:
Yes, this is the way it should be done. No magic inheritance. 
The goal is that layers as they are now are eventually made irrelevant and I know what I need to do to make that happen.